doc/proposals/composable-integration-in sdk.md

```diff
@@ -0,0 +1,173 @@
+For a detailed explanation of how to specify an object reference according to this schema, see [here](https://github.com/IBM/composable/blob/master/README.md#getvaluefrom-elements).
+
+The Composable SDK offers the following function for resolving the value of cross-resource references.
+
+```golang
+func ResolveObject(r client.Client, config *rest.Config, object interface{}, resolved interface{}, composableNamespace string) *ComposableError 
+```
+
+The function `ResolveObject` takes a Kubernetes client and configuration, and `object` to resolve, and a blank object
+`resolved` that will contain the result, as well as a namespace. The namespace is the default used when references
+do not specify one. This function will cast the result to the type of the `resolved` object, provided that
+appropriate data transformers have been included in the reference definitions (see [tutorial](https://github.com/IBM/composable/blob/master/sdk/docs/tutorial.md) for an example).
+
+The `ResolveObject` function uses caching for looking up object kinds, as well as for the objects themselves, in order
+to ensure that a consistent view of the data is obtained. If any data is not available at the time of the lookup,
+it returns an error. So this function either resolves the entire object or it doesn't -- there are no partial results.
+The cache lasts only for the duration of the `ResolveObject` call.
+
+Our proposal is to have a tight integration of `ResolveObject` into the API offered by the Operator-SDK to interact with
+Kubernetes objects.
+
+The return value is a `ComposableError`:
+
+```golang
+type ComposableError struct {
+	Error error
+	// This indicates that the consuming Reconcile function should return this error
+	ShouldBeReturned bool
+}
+```
+
+The type `ComposableError` contains the error, if any, and a boolean `ShouldBeReturned` to indicate whether the calling reconciler should return this error or not. This is used to distinguish errors that are minor and could be fixed by immediately retrying from errors that may indicate a stronger failure, such as an ill-formed yaml (in which case there is no need to retry right away). If there are no errors, then the `ResolveObject` returns nil.
+
+
+
+
+### Risks and Mitigations
+
+Using the Composable SDK means that the CRD needs to have permission to access objects that are being referenced.
+This means that the operator developer needs to consider what kinds of objects will be permitted for references and
+add appropriate permissions for the RBAC rules of the CRD.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+
+## Implementation History
+
+Composable SDK is currently available (version sdk/v0.1.3) at: https://github.com/IBM/composable/tree/master/sdk.
+
+
+## Drawbacks
+
```

Comment:
IHMO regards this subject this feature can have the potential to be very complex as well so we over how much value it will bring and how much effort it will reduce for developers. Note that, as described in https://github.com/operator-framework/operator-sdk/pull/2340/files#r366552692, we can just get any resource and update the CR with its values. So, if the users required to do a bigger implementation to achieve the same I am not sure how much value it could bring. 
c/c @joelanford